example/ring/ring.cpp

```diff
@@ -106,25 +113,37 @@ class ring_recipe: public arb::recipe {
 };
 
 struct cell_stats {
-    using size_type = std::uint64_t;
+    using size_type = unsigned;
     size_type ncells = 0;
     size_type nsegs = 0;
     size_type ncomp = 0;
 
-    cell_stats(arb::distributed_context_handle ctx, arb::recipe& r) {
-        size_type nranks = ctx->size();
-        size_type rank = ctx->id();
+    cell_stats(arb::recipe& r) {
+#ifdef ARB_MPI_ENABLED
+        int nranks, rank;
+        MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+        MPI_Comm_size(MPI_COMM_WORLD, &nranks);
```

Comment:
We have got wrappers in the implementation where all the MPI calls that we use internally where we error check, and do all the right things.
But we don't want to expose these to user code, because how applications that use Arbor handle their own MPI is their own business.
The applications in `example` are simple demonstrations of how to use Arbor. The point of having these direct MPI calls in the driving code is to clearly illustrate that it is user's responsibility to handle MPI on their own (yeah, I know, we provide the `with_mpi` wrapper).
The error handling isn't the best, but it is very clear, and correct so long as the application has been launched with `mpirun` or equivalent if necessary.